src/Command/AuditLogCleanupCommand.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Drupal\tide_core\Command;
+
+use Drush\Commands\DrushCommands;
+use Drupal\Core\Database\Database;
+use DateInterval;
+use DateTime;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Custom AuditLog Clean Drush command.
+ */
+class AuditLogCleanupCommand extends DrushCommands {
+
+/**
+   * The configuration service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Constructs a LogCleanupCommands object.
+   *
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The configuration service.
+   */
+  public function __construct(ConfigFactoryInterface $config_factory) {
+    $this->configFactory = $config_factory;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory')
+    );
+  }
+
+  /**
+   * Removes log entries older than the configured retention period.
+   *
+   * @command tide_core:auditlog-cleanup
+   * @aliases tcl
+   * @description Cleans up audittrail logs older than the configured retention period.
+   */
+  public function cleanupLogs() {
+    $config = $this->configFactory->get('tide_core.settings');
+    $log_retention_days = $config->get('log_retention_days') ?: 30;
```

Comment:
The default value for `$log_retention_days` is set inline. It would be clearer to define a constant for the default value.

Reply:
Fixed.